src/clinical/id-generator.ts

```diff
@@ -0,0 +1,293 @@
+ */
+
+import axios from 'axios';
+import memoize from 'memoizee';
+import ms from 'ms';
+import {
+	Biomarker,
+	ClinicalInfo,
+	Comorbidity,
+	CompletionStats,
+	Donor,
+	Exposure,
+	FamilyHistory,
+	FollowUp,
+	PrimaryDiagnosis,
+	SchemaMetadata,
+	Specimen,
+	Treatment,
+} from './clinical-entities';
+import { ClinicalEntitySchemaNames } from '../common-model/entities';
+import { config } from '../config';
+
+interface IdGenerationRequest {
+	programId: string;
+	submitterId: any;
+	submitterDonorId: string;
+	testInterval: string;
+	family_relative_id: string;
+	comorbidityTypeCode: string;
+	entityType: string;
+}
+
+const request = {
+	programId: 'null',
+	submitterId: 'null',
+	submitterDonorId: 'null',
+	testInterval: 'null',
+	family_relative_id: 'null',
+	comorbidityTypeCode: 'null',
+	entityType: 'null',
+};
+
+export interface PartialDonor {
+	_id?: string;
+	__v?: number; // mongodb property not being filtered out
+	createBy?: string;
+	schemaMetadata?: SchemaMetadata | undefined;
+	donorId?: number;
+	gender?: string;
+	submitterId?: string;
+	programId?: string;
+	specimens?: Array<Specimen> | undefined;
+	clinicalInfo?: ClinicalInfo;
+	primaryDiagnoses?: Array<PrimaryDiagnosis>;
+	familyHistory?: Array<FamilyHistory>;
+	comorbidity?: Array<Comorbidity>;
+	followUps?: Array<FollowUp>;
+	treatments?: Array<Treatment>;
+	exposure?: Array<Exposure>;
+	biomarker?: Array<Biomarker>;
+	createdAt?: string;
+	updatedAt?: string;
+	completionStats?: CompletionStats;
+}
+
+export async function getId(req: IdGenerationRequest) {
+	console.log('getId function called');
+	const token = await getToken();
+	const headers = {
+		headers: {
+			Authorization: 'Bearer ' + token,
+		},
+	};
+
+	try {
+		const response = await axios.get(
+			// `http://localhost:9001/${req.programId}/${req.submitterId}/${req.submitterDonorId}/${req.testInterval}/${req.family_relative_id}/${req.comorbidityTypeCode}/${req.entityType}`,
+			config.getConfig().tokenUrl() +
+				`${req.programId}/${req.submitterId}/${req.submitterDonorId}/${req.testInterval}/${req.family_relative_id}/${req.comorbidityTypeCode}/${req.entityType}`,
+			headers,
+		);
+		console.log('getId response: ' + response.data.entityId + ' - ' + response.data.entityType);
+		return parseInt(response.data.entityId);
+	} catch (e) {
+		console.log(e);
+		throw Error('Error fetching ids');
+	}
+}
+
+export async function setEntityIdsForDonors(donors: Donor[]) {
+	const donorsWithIds = donors.map(async (donor) => {
+		return await setEntityIds(donor);
+	});
+	return await Promise.all(donorsWithIds);
+}
+
+export async function setEntityIds(donor: PartialDonor) {
+	// -- DONOR --
+	const donorId = await getId({
+		...request,
+		programId: donor.programId as string,
+		submitterId: donor.submitterId as string,
+		submitterDonorId: donor.submitterId as string,
+		entityType: ClinicalEntitySchemaNames.DONOR,
+	});
+	donor.donorId = donorId;
+	console.log(donor.donorId);
```

Comment:
This should print more than just a number, it needs context, at the very least: 'donor.donorId: ` similar to the other logs in this function